File: programming/project-learning/r/maths/collatz-conjecture.r
number <- as.integer(readline(prompt = "Enter a number: "))  # Prompt the user to enter a number and convert it to an integer

while (number != 1) {  # Continue the loop until the number becomes 1
    if (number %% 2 == 0) {  # Check if the number is even
        number <- number %/% 2  # If even, divide the number by 2
    } else {  # If the number is odd
        number <- number * 3 + 1  # Multiply the number by 3 and add 1
    }
    print(number)  # Print the current value of the number
}

File: programming/project-learning/javascript/maths/collatz-conjecture.js
// Import the readline module to create an interface for reading input from the console
const readline = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

// Prompt the user to enter a number
readline.question("Enter a number: ", (number) => {
  // Convert the input to an integer
  number = parseInt(number);

  // Loop until the number becomes 1
  while (number !== 1) {
    // If the number is even, divide it by 2
    if (number % 2 === 0) {
      number = Math.floor(number / 2);
    } else {
      // If the number is odd, multiply it by 3 and add 1
      number = number * 3 + 1;
    }
    // Print the current value of the number
    console.log(number);
  }
  // Close the readline interface
  readline.close();
});
